### PR TITLE
Make LogRecord fields private and add getters for encapsulation

### PR DESCRIPTION
**Pull request by lalitb**
## Changes

The PR refactors the `LogRecord` struct by making all its fields crate-private for better abstraction. Public getter methods have been added to provide controlled access to the fields. 

TBD - Will add changelog, if the changes are agreed.

## Merge requirement checklist

* [x] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-rust/blob/main/CONTRIBUTING.md) guidelines followed
* [x] Unit tests added/updated (if applicable)
* [x] Appropriate `CHANGELOG.md` files updated for non-trivial, user-facing changes
* [x] Changes in public API reviewed (if applicable)

### CODE REVIEW

File: opentelemetry-sdk/src/logs/record.rs [hunk 1]
```diff
@@ -118,7 +118,56 @@ impl opentelemetry::logs::LogRecord for LogRecord {
...
+    #[inline]
+    pub fn event_name(&self) -> Option<&'static str> {
+        self.event_name
+    }
+
+    /// Returns the target
+    #[inline]
+    pub fn target(&self) -> Option<&Cow<'static, str>> {
+        self.target.as_ref()
+    }
```
Comment (lalitb) [inline opentelemetry-sdk/src/logs/record.rs:131]: Returning &str would simplify the API, but want to preserve the ability for consumers to optimize based on whether the target is a static string.

Comment (utpilla) [reply opentelemetry-sdk/src/logs/record.rs:131]: I think we could favor simplicity of the API here and use `&str` as the return type. The current exporters that we have cannot utilize the `static` str reference to their benefit anyway.

Only exporters that would benefit from `Cow<'static, str>` would be the ones that hold these values in memory for some reason as they could then avoid the cost to clone a static str.

**Review by utpilla — APPROVED** [review]

File: opentelemetry-sdk/src/logs/record.rs (hunk 1 above)
Comment (utpilla) [reply opentelemetry-sdk/src/logs/record.rs:131]: Not a strong opinion here as it's mainly used by exporters and not end users. We can expect exporter authors to work with `Cow<'static,str>`.

Comment (lalitb) [reply opentelemetry-sdk/src/logs/record.rs:131]: > Not a strong opinion here as it's mainly used by exporters and not end users. We can expect exporter authors to work with Cow<'static,str>.

Yes, that's what I thought - keeping it for the custom export authors if they want to reap the benefit of static target.

**Review by cijothomas — APPROVED** [review]
LGTM with changelog as this is breaking export/processor authors.

**Comment by lalitb** [comment]
> LGTM with changelog as this is breaking export/processor authors.

updated changelog.

**Pull request merged**